### ISSUE TITLE
[REVIEW]: Mordecai: Full Text Geoparsing and Event Geocoding

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#91]
**Submitting author:** @ahalterman (<a href="http://orcid.org/0000-0001-9716-9555">Andrew Lee Halterman</a>)
**Repository:** <a href="https://github.com/openeventdata/mordecai" target ="_blank">https://github.com/openeventdata/mordecai</a>
**Version:** v1.1.0
**Editor:** @acabunoc
**Reviewer:** @riordan
**Archive:** <a href="http://dx.doi.org/10.5281/zenodo.250879" target="_blank">10.5281/zenodo.250879</a>

## Status

[![status](http://joss.theoj.org/papers/20e4b21d74e3ba78d2fb2fde5403be96/status.svg)](http://joss.theoj.org/papers/20e4b21d74e3ba78d2fb2fde5403be96)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/20e4b21d74e3ba78d2fb2fde5403be96"><img src="http://joss.theoj.org/papers/20e4b21d74e3ba78d2fb2fde5403be96/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/20e4b21d74e3ba78d2fb2fde5403be96/status.svg)](http://joss.theoj.org/papers/20e4b21d74e3ba78d2fb2fde5403be96)
```

**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/openeventdata/mordecai" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)
### Conflict of interest
- [x] As the reviewer I confirm that there are no conflicts of interest for me to review this work (such as being a major contributor to the software).
### General checks
- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/openeventdata/mordecai">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** Does the release version given match the GitHub release (v1.1.0)?
- [x] **Authorship:** Has the submitting author (@ahalterman) made major contributions to the software?
### Functionality
- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** Have any performance claims of the software been confirmed?
### Documentation
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g. API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support
### Software paper

Paper PDF: [10.21105.joss.00091.pdf](https://github.com/openjournals/joss-reviews/files/525307/10.21105.joss.00091.pdf)
- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g. papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon. I'm here to help you with some common editorial tasks for JOSS.

For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by acabunoc** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List all of Whedon's capabilities
@whedon commands

# Assign a GitHub user as the reviewer of this submission
@whedon assign @username as reviewer

# List the GitHub usernames of the JOSS editors
@whedon list editors

# List of JOSS reviewers together with programming language preferences and domain expertise
@whedon list reviewers

# Change editorial assignment
@whedon assign @username as editor

# Set the software archive DOI at the top of the issue e.g.
@whedon set 10.0000/zenodo.00000 as archive

# Open the review issue
@whedon start review
```

:construction: **Important** :construction:

This is all quite new. Please make sure you check the top of the issue after running a @whedon command (you might also need to refresh the page to see the issue update).

**Comment by ahalterman** [comment]
@riordan, let me know if you run into any trouble with the documentation/installation. The model's pretty big, unfortunately, so getting it stood up doesn't go smoothly depending on how much memory you have.

**Comment by riordan** [comment]
@whedon start review

**Comment by whedon** [comment]
I'm sorry @riordan, I'm afraid I can't do that. That's something only JOSS editors are allowed to do.

**Comment by arfon** [comment]
@riordan - need any help? Looks like the review is already happening (you're checking items off the list)

**Comment by ahalterman** [comment]
I merged a PR in that reorganizes how it deals with the data download, which is something you had changed, @riordan. This is sort of a general JOSS question, should I re-release the repo and change which version is being submitted? Or should we keep going with the existing version? The differences aren't too big so it shouldn't matter too much either way.

**Comment by arfon** [comment]
> This is sort of a general JOSS question, should I re-release the repo and change which version is being submitted? Or should we keep going with the existing version? The differences aren't too big so it shouldn't matter too much either way.

@ahalterman - I would recommend making a new release once the review is complete and all of the changes (associated with the review) have been made.

**Comment by ahalterman** [comment]
I wanted to check in and see if there's anything I can do to help along the review process. I really appreciate all the work you all have already put in.

**Comment by riordan** [comment]
@ahalterman: Thank you for the prod! I finished my tests up 2 weeks ago and it's really impressive.

The installation ran according to documentation, which was really straightforward for installation. I'd suggest a clearer statement of need in the paper (something you do well in the README.md). Bringing in a bit more context, especially from your experience scanning news articles to find the places that are being written about, would be really useful here.

Other than that, I think it just needs an archival DOI (though I'd defer to @acabunoc or @arfon on that).

**Comment by ahalterman** [comment]
@riordan Thanks so much for doing this review! I'm glad you liked it and that things checked out. I just updated the paper to include more of the statement of need. If it looks good to you, it sounds like the paper ready to go. Thanks everyone!

**Comment by ahalterman** [comment]
I wanted to check in again and see if that commit addressed your last request, @riordan. If so, I think it should be ready for a DOI.

**Comment by riordan** [comment]
Totally! 👍

**Comment by ahalterman** [comment]
Great! Thanks. I owe you a beer next time I'm in New York or you're in Cambridge. @acabunoc or @arfon, what happens now?

**Comment by acabunoc** [comment]
Thanks for the great work here @riordan & @ahalterman 👏  🎉 

@ahalterman, could you make sure there's an archive of the reviewed software in Zenodo/figshare/other service and update this thread with the DOI of the archive? Thanks!

**Comment by ahalterman** [comment]
Here it is! Thanks a lot.

[![DOI](https://zenodo.org/badge/44676568.svg)](https://zenodo.org/badge/latestdoi/44676568)

**Comment by ahalterman** [comment]
Here it is! Thanks a lot.

[![DOI](https://zenodo.org/badge/61832342.svg)](https://zenodo.org/badge/latestdoi/61832342)

**Comment by acabunoc** [comment]
@whedon set 10.5281/zenodo.250879 as archive

**Comment by whedon** [comment]
OK. <a href="http://dx.doi.org/10.5281/zenodo.250879" target="_blank">10.5281/zenodo.250879</a> is the archive.

**Comment by acabunoc** [comment]
flagging @arfon, I think we're good to go here! Thanks @riordan & @ahalterman ✨

**Comment by arfon** [comment]
@ahalterman your paper is now accepted into JOSS and your paper DOI is http://dx.doi.org/10.21105/joss.00091 ⚡️ 🚀 💥
